referenceframe/frame_system.go

```diff
@@ -308,6 +309,67 @@ func (sfs *simpleFrameSystem) FrameSystemSubset(newRoot Frame) (FrameSystem, err
 	return newFS, nil
 }
 
+// FrameSystemToPCD takes in a framesystem and returns a map where all elements are
+// the point representation of their geometry type with respect to the world.
+func FrameSystemToPCD(system FrameSystem, inputs map[string][]Input) (map[string][]r3.Vector, error) {
+	vectorMap := make(map[string][]r3.Vector)
+	geoMap, err := FrameSystemGeometries(system, inputs)
+	if err != nil {
+		return nil, err
+	}
+	for name, geosInFrame := range geoMap {
+		geos := geosInFrame.geometries
+		aggregatePoints := []r3.Vector{}
+		for _, g := range geos {
+			asPoints := g.ToPoints(1.)
+			aggregatePoints = append(aggregatePoints, asPoints...)
+		}
+		if len(aggregatePoints) > 0 {
```

Comment:
suppose the following.
we have an arm and a gripper.
Then `FrameSystemGeometries` will return a map with keys: 'arm`, `arm_origin`, 'gripper`, and 'gripper_origin`.
Where `arm_origin`.`ToPoints` == `gripper`.`ToPoints` == `0`.
This is why I added this conditional check.
However, it makes more sense to have this conditional exist further upstream. I think it would be most efficient to add a conditional on line 321 which check if the volume of the geometry is not non-zero. This in turn would require the addition of the `Volume()` method to the `Geometry` interface. Lmk what you think

Reply:
edit: spoke with ray offline.
This should not be a check that we care about. I will try to reproduce an example where this happens, but if I am unable to, then I will remove this check. Update coming in a bit.